tests/json-fixtures/test_transactions.py

```diff
@@ -40,68 +45,90 @@
+    elif fork_name == ForkName.Constantinople:
+        pytest.skip("Constantinople Transaction class has not been implemented")
+    elif fork_name == ForkName.Metropolis:
+        pytest.skip("Metropolis Transaction class has not been implemented")
+    else:
+        raise ValueError("Unknown Fork Name: {0}".format(fork_name))
 
-    if 'sender' in fixture:
-        transaction = rlp.decode(fixture['rlp'], sedes=TransactionClass)
-        expected = normalize_signed_transaction(fixture['transaction'])
-
-        assert transaction.nonce == expected['nonce']
-        assert transaction.gas_price == expected['gasPrice']
-        assert transaction.gas == expected['gasLimit']
-        assert transaction.to == expected['to']
-        assert transaction.value == expected['value']
-        assert transaction.data == expected['data']
-        assert transaction.v == expected['v']
-        assert transaction.r == expected['r']
-        assert transaction.s == expected['s']
 
-        sender = to_canonical_address(fixture['sender'])
+def test_transaction_fixtures(fixture, fixture_transaction_class):
 
-        try:
-            assert transaction.sender == sender
-        except BadSignature:
-            assert not (27 <= transaction.v <= 34)
-    else:
-        # check RLP correctness
-        try:
-            transaction = rlp.decode(fixture['rlp'], sedes=TransactionClass)
-        # fixture normalization changes the fixture key from rlp to rlpHex
-        except KeyError:
-            assert fixture['rlpHex']
-            return
-        # rlp is a list of bytes when it shouldn't be
-        except TypeError as err:
-            assert err.args == ("'bytes' object cannot be interpreted as an integer",)
-            return
-        # rlp is invalid or not in the correct form
-        except (rlp.exceptions.ObjectDeserializationError, rlp.exceptions.DecodingError):
-            return
+    TransactionClass = fixture_transaction_class
 
+    try:
+        txn = rlp.decode(fixture['rlp'], sedes=TransactionClass)
+    except (rlp.DeserializationError, rlp.exceptions.DecodingError):
+        assert 'hash' not in fixture, "Transaction was supposed to be valid"
+    except TypeError as err:
+        assert err.args == ("'bytes' object cannot be interpreted as an integer",)
+        assert 'hash' not in fixture, "Transaction was supposed to be valid"
+    # fixture normalization changes the fixture key from rlp to rlpHex
+    except KeyError:
+        assert fixture['rlpHex']
+        assert 'hash' not in fixture, "Transaction was supposed to be valid"
+    else:
         # check parameter correctness
         try:
-            transaction.validate()
+            txn.validate()
         except ValidationError:
             return
+
+    if 'sender' in fixture:
+        assert 'hash' in fixture, "Transaction was supposed to be invalid"
+        sender = to_canonical_address(fixture['sender'])
+
+        try:
+            assert txn.get_sender() == sender
```

Comment:
No need to change this here but I wanted to be sure you knew about [`eth_utils.is_same_address`](https://github.com/ethereum/eth-utils#is_same_addressa-b---bool) for cases where you need to compare addresses in different forms.

Reply:
Thanks for bringing that up!

Reply:
I actually updated the code to use it while I was already at it.